### PR TITLE
Support setting output verbosity

### LINKED ISSUE #594: Suppress watchmedo on_any_event log messages
When using watchmedo auto-restart how can I suppress all the event messages logged to the console? It's noise I do not want to see. 

Thanks.

Eg:
```shell
$ watchmedo auto-restart test.sh
on_any_event(self=<watchdog.tricks.AutoRestartTrick object at 0xb634ab30>, event=<FileCreatedEvent: src_path='./4913'>)
on_any_event(self=<watchdog.tricks.AutoRestartTrick object at 0xb634ab30>, event=<DirModifiedEvent: src_path='.'>)
on_any_event(self=<watchdog.tricks.AutoRestartTrick object at 0xb634ab30>, event=<FileModifiedEvent: src_path='./4913'>)
on_any_event(self=<watchdog.tricks.AutoRestartTrick object at 0xb634ab30>, event=<FileDeletedEvent: src_path='./4913'>)
on_any_event(self=<watchdog.tricks.AutoRestartTrick object at 0xb634ab30>, event=<DirModifiedEvent: src_path='.'>)
```

### PR DESCRIPTION
**Pull request by taleinat**
This changes the internal use of echo to use a logger instead of `sys.stdout.write`.

This uses the stdlib logging library, with the common convention of `logger = logging.getLogger(__name__)`, so that all watchdog logs are under the "watchdog" logging context. This allows code using watchdog to easily change the logging level of watchdog code via e.g. `logging.getLogger("watchdog").setLevel("ERROR")`.

Finally, this adds -q/--quiet and -v/--verbose command line options to watchmedo.

Fixes #594.

P.S. This also makes watchmedo exit with a nonzero return code when a command is not provided or when -v or -q are supplied too many times.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Thanks @taleinat :)

When running `watchmedo` without `-q|-v`, I got:

```python
Traceback (most recent call last):
  File "/bin/watchmedo", line 33, in <module>
    sys.exit(load_entry_point('watchdog', 'console_scripts', 'watchmedo')())
  File "src/watchdog/watchmedo.py", line 650, in main
    verbosity = sum(args.verbosity)
TypeError: 'NoneType' object is not iterable
```

And would you mind adding some documentation in the README + a line in the changelog?

**Comment by taleinat** [comment]
Done as suggested, @BoboTiG.

Regarding adding to the README, it currently gives two examples and refers the reader to running `--help`, rather than including a comprehensive description of all command line options. Given that, I think expanding on verbosity there would require more rewriting than I think is prudent in this context.

**Comment by BoboTiG** [comment]
You are right :+1:

**Comment by BoboTiG** [comment]
I am OK to merge if you are done.

**Comment by taleinat** [comment]
I'm done, so if you've reviewed the latest changes and are okay with them, let's merge :)

File: src/watchdog/watchmedo.py
Comment (BoboTiG) [inline src/watchdog/watchmedo.py:663]: I think one thing is missing: where did you populate `command_parsers`?

```bash
$ watchmedo log -vvv\
    --patterns="*.py;*.txt" \
    --ignore-directories \
    --recursive \
    .
```
```python
Error: -v/--verbose may be specified up to 2 times.
Traceback (most recent call last):
  File "src/watchdog/watchmedo.py", line 660, in main
    log_level = _get_log_level_from_args(args)
  File "src/watchdog/watchmedo.py", line 648, in _get_log_level_from_args
    raise Exception("-v/--verbose may be specified up to 2 times.")
Exception: -v/--verbose may be specified up to 2 times.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "venv3.9.8/bin/watchmedo", line 33, in <module>
    sys.exit(load_entry_point('watchdog', 'console_scripts', 'watchmedo')())
  File "src/watchdog/watchmedo.py", line 663, in main
    command_parsers[args.top_command].print_help()
KeyError: 'log'
```

**Review by BoboTiG — COMMENTED** [review]
Last minute check :)

File: src/watchdog/watchmedo.py
Comment (BoboTiG) [reply src/watchdog/watchmedo.py:663]: And maybe should we raise a custom exception rather than the broad `Exception`. WDYT?

Comment (taleinat) [reply src/watchdog/watchmedo.py:664]: Whoops, good catch! That was indeed missing, not sure how it got dropped.

**Comment by taleinat** [comment]
Please wait before merging...

**Comment by taleinat** [comment]
I've made the code use a specific custom exception rather than raising Exception. Ready now as far as I'm concerned.

**Review by BoboTiG — APPROVED** [review]

**Comment by BoboTiG** [comment]
Perfect! Thanks a lot :champagne: :confetti_ball:

**Comment by taleinat** [comment]
Excellent!

Any plans for a release soon?

**Comment by BoboTiG** [comment]
Well, right now seems a good time ;)

**Pull request merged**
